tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java

```diff
@@ -62,6 +62,7 @@ public boolean rollbackOn(Throwable ex) {
         int deepest = Integer.MAX_VALUE;
 
         if (this.rollbackRules != null) {
+            winner = NoRollbackRule.DEFAUL_NO_ROLLBACK_RULE;
```

Comment:
the rollbackRules is not null but it is empty, not do roll back?

Reply:
> the rollbackRules is not null but it is empty, not do roll back?

When might it be empty?

Reply:
![image](https://user-images.githubusercontent.com/26616196/67366629-6f2b5d80-f5a6-11e9-918d-5c2adf44c6fc.png)

Reply:
OK.Then this.rollbackRules != null should be replaced with CollectionUtils.isNotEmpty.

Reply:
done ,thanks for @jsbxyyx